pkg/solana/client/client.go

```diff
@@ -82,75 +118,207 @@ func (h *Head) BlockNumber() int64 {
 }
 
 func (h *Head) BlockDifficulty() *big.Int {
+	// Not relevant for Solana
 	return nil
 }
 
 func (h *Head) IsValid() bool {
-	return h.BlockHeight != nil
-}
-
-func NewClient(endpoint string, cfg config.Config, requestTimeout time.Duration, log logger.Logger) (*Client, error) {
-	return &Client{
-		url:             endpoint,
-		rpc:             rpc.New(endpoint),
-		skipPreflight:   cfg.SkipPreflight(),
-		commitment:      cfg.Commitment(),
-		maxRetries:      cfg.MaxRetries(),
-		txTimeout:       cfg.TxTimeout(),
-		contextDuration: requestTimeout,
-		log:             log,
-		requestGroup:    &singleflight.Group{},
-	}, nil
+	return h.BlockHeight != nil && h.BlockHash != nil
 }
 
 var _ mn.RPCClient[mn.StringID, *Head] = (*Client)(nil)
 var _ mn.SendTxRPCClient[*solana.Transaction] = (*Client)(nil)
 
-// TODO: BCI-4061: Implement Client for MultiNode
-
 func (c *Client) Dial(ctx context.Context) error {
-	//TODO implement me
-	panic("implement me")
+	return nil
 }
 
 func (c *Client) SubscribeToHeads(ctx context.Context) (<-chan *Head, mn.Subscription, error) {
-	//TODO implement me
-	panic("implement me")
+	if c.pollInterval == 0 {
+		return nil, nil, errors.New("PollInterval is 0")
+	}
+	timeout := c.pollInterval
+	poller, channel := mn.NewPoller[*Head](c.pollInterval, c.LatestBlock, timeout, c.log)
```

Comment:
poller should be registered as subscription. Otherwise Node's state transition won't cause it and won't force application layer to reconnect